promql/engine_test.go

```diff
@@ -1039,7 +1039,7 @@ load 10s
 			Start:        time.Unix(201, 0),
 			End:          time.Unix(220, 0),
 			Interval:     5 * time.Second,
-			PeakSamples:  5,
+			PeakSamples:  8,
```

Comment:
note for reviewers: 8 because
- 4 samples of `metricWith1SampleEvery10Seconds` loaded at `ts 201, 206, 211, 216`
- 4 samples from 4 calls to `funcTimestamp` used to construct the response

Reply:
> but those four copies are not kept in memory at the same time, but only one at a time.

I think this is incorrect and they are all stored in memory at the same time to construct the response matrix, see [1243](https://github.com/prometheus/prometheus/pull/9626/files#diff-ad2761bb162a2f12326b18f2714ef54f4cd476cfe3388f77b2b20c5ff82e4faaR1243)